boringtun/src/device/peer.rs

```diff
@@ -63,31 +65,31 @@ impl Peer {
         preshared_key: Option<[u8; 32]>,
         protect: Arc<dyn MakeExternalBoringtun>,
     ) -> Peer {
+        let pub_key = tunnel.peer_static_public();
+        let mut public_key_hex = String::with_capacity(32);
+        for byte in pub_key.as_bytes() {
+            let pub_symbol = format!("{:02X}", byte);
+            public_key_hex.push_str(&pub_symbol);
+        }
```

Comment:
nit: boringtun already depends on hex, so I think here you can use [hex::encode](https://docs.rs/hex/0.4.3/hex/fn.encode.html).